pkg/daemon/daemon.go

```diff
@@ -208,62 +214,55 @@ func New(
 		glog.Infof("Booted osImageURL: %s (%s)", osImageURL, osVersion)
 	}
 
-	bootID, err := getBootID()
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to read boot ID")
-	}
-
-	dn := &Daemon{
-		booting:                true,
-		name:                   nodeName,
-		OperatingSystem:        operatingSystem,
-		NodeUpdaterClient:      nodeUpdaterClient,
-		bootedOSImageURL:       osImageURL,
-		bootID:                 bootID,
-		kubeletHealthzEnabled:  kubeletHealthzEnabled,
-		kubeletHealthzEndpoint: kubeletHealthzEndpoint,
-		nodeWriter:             nodeWriter,
-		exitCh:                 exitCh,
-		stopCh:                 stopCh,
-		kubeClient:             kubeClient,
-		mcClient:               mcClient,
-	}
-	dn.currentConfigPath = currentConfigPath
-	dn.atomicSSHKeysWriter = dn.atomicallyWriteSSHKey
-	dn.loggerSupportsJournal = dn.isLoggingToJournalSupported()
-	return dn, nil
+	bootID := ""
+	if !mock {
+		bootID, err = getBootID()
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to read boot ID")
+		}
+	}
+
+	// RHEL 7.6 logger (util-linux) doesn't have the --journald flag
+	loggerSupportsJournal := true
+	if !mock {
+		if operatingSystem == machineConfigDaemonOSRHEL {
```

Comment:
can we just have like before:

```
if operatingsystem == RHCOS { return true }
else
check --journald
```

Reply:
(where the bug was `s/OSRHEL/RHCOS`)

Reply:
The code is now in the constructor so we can't `return true`.